R/assertions.R

```diff
@@ -330,9 +394,15 @@ insist <- function(data, predicate_generator, ...,
                       return(apply.predicate.to.vector(this.vector,
                                                        predicate))})
 
-  # if all checks pass *and* there are no leftover errors
-  if(all(log.mat) && is.null(attr(data, "assertr_errors")))
-    return(success_fun(data))
+  # if all checks pass in current assertion
+  if(all(log.mat))
+    return(success_fun(data, "insist", name.of.predicate.generator, colnames(sub.frame), NA, description))
+
+  # if errors occured and verification was obligatory
+  if(obligatory)
+    attr(data, "assertr_data_defective") <- TRUE
+
+  assertion.id <- generate_id()
```

Comment:
I'm pretty confused with naming convention. I don't have strong preferences to any, but until this point I assumed that dotted (`a.b`) is for functions and snaked (`a_b`) for variables, but it doesn't seem to be a case here ?

Reply:
I tried to find out naming convention in this package. The things I realized are:
* not exported functions and S3 methods are dotted
* exported functions are snaked
* variables passed (or the one that supposes to be passed) to any of above type of methods inherits the style from the method.

In example in here assertion.id is passed to  `make.assertr.assert.error` (that is not exported function) and all parameters passed to it are dotted.